trait2d/analysis/__init__.py

```diff
@@ -471,23 +473,65 @@ def adc_summary(self, avg_only_params = False, interpolation = False, plot_msd =
         for model in counter:
             sector[model] = counter[model] / len(self._tracks)
         sector["not catergorized"] = (len(self._tracks) - counter_sum) / len(self._tracks)
+        
+        ###ENSEMBLE AVERAGE - Added by FR###
+        
+        ens_D_app = np.zeros(track_length - 3)
+        ens_MSD = np.zeros(track_length - 3)
+        
+        if ensemble_average:
+            for track in self._tracks:
+                if track.get_adc_analysis_results() is None:
+                    continue
+                
+                D_app = np.zeros(track_length - 3)
+                MSD = np.zeros(track_length - 3)
+                if interpolation:
+                    interp_MSD = interpolate.interp1d(track.get_t()[0:-3], track.get_msd(), bounds_error = False, fill_value = 0)
+                    interp_D_app = interpolate.interp1d(track.get_t()[0:-3], track.get_adc_analysis_results()["Dapp"], bounds_error = False, fill_value = 0)
+                    MSD = interp_MSD(t[0:-3])
+                    D_app = interp_D_app(t[0:-3])
+                else:
+                    D_app[0:track.get_adc_analysis_results()["Dapp"].size] = track.get_adc_analysis_results()["Dapp"]
+                    MSD[0:track.get_msd().size] = track.get_msd()
+                
+                ens_D_app += D_app
+                ens_MSD += MSD
+                
+            average_MSD['Ensemble'] = ens_MSD/len(self._tracks)
+            average_D_app['Ensemble'] = ens_D_app/len(self._tracks)
```

Comment:
This might be problematic as the number of tracks is not necessarily the number of tracks used for averaging (e.g. if no ADC results have been found for a track or a track is too short).

```suggestion
            mask = np.zeros(track_length - 3)
            np.put(mask, np.where(MSD != 0.0), 1)
            
            average_D_app['Ensemble'] += D_app
            average_MSD['Ensemble'] += MSD
            sampled += mask
            
            average_MSD['Ensemble'] = ens_MSD/sampled
            average_D_app['Ensemble'] = ens_D_app/sampled
```